include/fmt/format.h

```diff
@@ -3544,8 +3544,14 @@ FMT_CONSTEXPR internal::udl_arg<wchar_t> operator"" _a(const wchar_t* s,
 #endif  // FMT_USE_USER_DEFINED_LITERALS
 FMT_END_NAMESPACE
 
+#if FMT_GCC_VERSION >= 840
+#  define FMT_STRING_RESULT(s)   ()->fmt::basic_string_view<fmt::remove_cvref_t<decltype(*s)>>
```

Comment:
I think this will break compile-time string detection: https://github.com/fmtlib/fmt/blob/1e8493196efa4e05baa22461ba61b636ab2e662b/include/fmt/core.h#L460